domains/client/block-preprocessor/src/lib.rs

```diff
@@ -306,6 +301,19 @@ where
                 )));
             }
 
+            // Check if this extrinsic is an inherent extrinsic.
+            // If so, this is an invalid bundle since these extrinsics should not be included in the
+            // bundle. Extrinsic is always decodable due to the check above.
+            let is_inherent_extrinsic = runtime_api
+                .is_inherent_extrinsic(at, extrinsic.encode())?
+                .unwrap_or(true);
```

Comment:
I'm confused, is there any specific reason that requires encoded extrinsic?

Reply:
This is infra used to pass this encoded extrinsic while verifying the fraud proof. So passed in as encoded one so that domain runtime itself can decode it accordingly rather than deriving `Block::Extrinsic`.

Reply:
The `extrinsic` is already decoded, it is just a waste to encode and decode it again. 

> rather than deriving Block::Extrinsic

Not sure what you mean, but other runtime api/fraud proof already using `Block::Extrinsic` (i.e. `is_within_tx_range`, `check_transaction_validity`) doesn't see it bring any obvious downside.